src/cart/crt_hg.c

```diff
@@ -1237,7 +1237,12 @@ crt_rpc_handler_common(hg_handle_t hg_hdl)
 		  rpc_priv->crp_opc_info->coi_opc,
 		  &rpc_priv->crp_pub);
 
-	crt_rpc_priv_init(rpc_priv, crt_ctx, true /* srv_flag */);
+	rc = crt_rpc_priv_init(rpc_priv, crt_ctx, true /* srv_flag */);
+	/* will fail if rpc deadline expired */
```

Comment:
I"m not a big fan of having priv_init fail, all that function should do is initalize the private data. Now it does some timeout calculation inside, it's a bit misleading in my opinion. Maybe it would be best to separate the deadline stuff from priv_init ?